cpu/esp32/include/adc_arch.h

```diff
@@ -108,14 +145,19 @@ int adc_line_vref_to_gpio(adc_t line, gpio_t gpio);
  *
  * This function is deprecated and will be removed in future versions.
  *
+ * @warning The function is not supported any longer, use
+ *          @ref adc_raw_to_voltage to get the voltage for a sample value.
+ *
  * @return  0 on success
  * @return  -1 on invalid ADC line
  */
+__attribute__((__deprecated__))
```

Comment:
I do like deprecation in code a lot.

@chrysn: I recall that you started a PR to have deprecation in both code and doc. Was there any reason that this did not get in yet, or just time constraints?

Specifically: Would any of the issues that you may have run into be worth watching out for, or can we just start to gradually use deprecation in code?